flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -219,25 +278,38 @@ private void flush() {
         List<RequestEntryT> batch = new ArrayList<>(maxBatchSize);
 
         int batchSize = Math.min(maxBatchSize, bufferedRequestEntries.size());
+        int batchSizeBytes = 0;
         for (int i = 0; i < batchSize; i++) {
             RequestEntryWrapper<RequestEntryT> elem = bufferedRequestEntries.remove();
             batch.add(elem.getRequestEntry());
             bufferedRequestEntriesTotalSizeInBytes -= elem.getSize();
+            batchSizeBytes += elem.getSize();
         }
 
         if (batch.size() == 0) {
             return;
         }
 
+        long timestampOfRequest = System.currentTimeMillis();
         Consumer<Collection<RequestEntryT>> requestResult =
                 failedRequestEntries ->
                         mailboxExecutor.execute(
-                                () -> completeRequest(failedRequestEntries),
+                                () -> completeRequest(failedRequestEntries, timestampOfRequest),
                                 "Mark in-flight request as completed and requeue %d request entries",
                                 failedRequestEntries.size());
 
+        Consumer<Exception> fatalExceptionCons =
```

Comment:
Can we make this an instance variable and only instantiate it once? AFAICT it does not hold a reference to a variable of this method?

Reply:
If it is a field we can probably also expose it to subclasses directly and do not need to change `submitRequestEntries` wdyt?

Reply:
Ah nice, that's even cooler.

Reply:
Done :D